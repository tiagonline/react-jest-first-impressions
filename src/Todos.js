import React, { Component } from 'react';

import { connect } from 'react-redux';

class Todos extends Component {
  render() {
    return (
        <ul className="Todos">
            { this.props.todos.map(todo => (
                <li key={todo.id}>{todo.text}</li>
            )) }
        </ul>
    );
  }
}

//acessa o reducers de todos
const mapStateToProps = state => ({
    todos: state.todos,
});

export default connect(mapStateToProps)(Todos);
