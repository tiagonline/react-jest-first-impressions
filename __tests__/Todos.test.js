import React from 'React';
import { shallow } from 'enzyme';
import Todos from '../src/Todos';

// mock o estado do redux
import configureStore from 'redux-mock-store';
const mockStore = configureStore();

//cria estado inicial de todos mockados
const initialState = {
    todos : [
        {id: 0, text: 'Fazer café'},
        {id: 1, text: 'Fazer xá'},
        {id: 2, text: 'Fazer chocolate'},
    ],
};  

const store = mockStore(initialState);

describe('Testing Todos Component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <Todos/>,
            { context: {store} }
        ).dive();
        //Shallow render the one non-DOM child of the current wrapper, and return a wrapper around the result.
        
        expect(wrapper).toMatchSnapshot();   
    });
});