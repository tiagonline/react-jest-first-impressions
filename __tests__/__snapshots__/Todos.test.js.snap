// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todos
    dispatch={[Function]}
    todos={
        Array [
            Object {
              "id": 0,
              "text": "Fazer café",
            },
            Object {
              "id": 1,
              "text": "Fazer xá",
            },
            Object {
              "id": 2,
              "text": "Fazer chocolate",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          Fazer café
</li>,
        <li>
          Fazer xá
</li>,
        <li>
          Fazer chocolate
</li>,
      ],
      "className": "Todos",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Fazer café",
        },
        "ref": null,
        "rendered": "Fazer café",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Fazer xá",
        },
        "ref": null,
        "rendered": "Fazer xá",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Fazer chocolate",
        },
        "ref": null,
        "rendered": "Fazer chocolate",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            Fazer café
</li>,
          <li>
            Fazer xá
</li>,
          <li>
            Fazer chocolate
</li>,
        ],
        "className": "Todos",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Fazer café",
          },
          "ref": null,
          "rendered": "Fazer café",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Fazer xá",
          },
          "ref": null,
          "rendered": "Fazer xá",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Fazer chocolate",
          },
          "ref": null,
          "rendered": "Fazer chocolate",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
